### PR TITLE
EZP-29041: Content still visible in ezobjectrelationlist after sending it to trash

### PR DESCRIPTION
**Pull request by mikadamczyk**
| Question           | Answer
| ------------------ | ------------------
| **JIRA issue**     | [EZP-29041](https://jira.ez.no/browse/EZP-29041)
| **Bug**| yes
| **New feature**    |no
| **Target version** | `7.3`
| **BC breaks**      |no
| **Tests pass**     | yes
| **Doc needed**     |no

**TODO**:
- [x] Implement feature / fix a bug.
- [x] Implement tests.
- [ ] Fix new code according to Coding Standards (`$ composer fix-cs`).
- [ ] Ask for Code Review.

### CODE REVIEW

File: eZ/Publish/Core/Persistence/Cache/TrashHandler.php [hunk 1]
```diff
@@ -99,7 +99,25 @@ public function findTrashItems(Criterion $criterion = null, $offset = 0, $limit
...
         $this->logger->logCall(__METHOD__, array());
+
+        $trashedItems = $this->persistenceHandler->trashHandler()->findTrashItems();
+
+        $tags = [[]];
+        foreach ($trashedItems as $trashedItem) {
+            $contentId = $trashedItem->contentId;
+            $reverseRelations = $this->persistenceHandler->contentHandler()->loadReverseRelations($contentId);
+
+            if (!empty($reverseRelations)) {
```
Comment (andrerom) [inline eZ/Publish/Core/Persistence/Cache/TrashHandler.php]: given it always returns array we can skip this if right?
But we should probably move it to `invalidateTags()` so we don't call that with empty tags array.

File: eZ/Publish/Core/Persistence/Cache/TrashHandler.php [hunk 2]
```diff
@@ -99,7 +99,25 @@ public function findTrashItems(Criterion $criterion = null, $offset = 0, $limit
...
         $this->logger->logCall(__METHOD__, array());
+
+        $trashedItems = $this->persistenceHandler->trashHandler()->findTrashItems();
+
+        $tags = [[]];
+        foreach ($trashedItems as $trashedItem) {
+            $contentId = $trashedItem->contentId;
+            $reverseRelations = $this->persistenceHandler->contentHandler()->loadReverseRelations($contentId);
+
+            $tags[] = array_map(function (Relation $relation) {
```
Comment (adamwojs) [inline eZ/Publish/Core/Persistence/Cache/TrashHandler.php]: You can avoid array_merge by replacing array_map with array_walk (or just inner loop).

File: eZ/Publish/Core/Persistence/Cache/TrashHandler.php (hunk 1 above)
Comment (mikadamczyk) [reply eZ/Publish/Core/Persistence/Cache/TrashHandler.php]: Thanks @andrerom, it is fixed now

**Comment by mikadamczyk** [comment]
@alongosz, we can not use the return value of emptyTrash method because, in the next step, we are not able to fetch the reverse relations of deleted content.

**Comment by andrerom** [comment]
Well... We could add tags to allow us to clear based on that info, it's a matter of reverse relationships right?

So current item, needs to have tags for all it's relations, so that when they change they can clear reverse relations.

**Comment by lserwatka** [comment]
You can merge it up.

**Pull request merged**